erniebot-agent/applications/erniebot_researcher/group_agent.py

```diff
@@ -0,0 +1,265 @@
+import logging
```

Comment:
这个文件里的代码有用到吗? 我没看到在哪里使用了

Reply:
在这里：

```
    def manual_select_speaker(self, agents: List[Agent]) -> Union[Agent, None]:
        logger.info("请从以下列表中选择下一位Agent：")
        _n_agents = len(agents)
        for i in range(_n_agents):
            logger.info(f"{i+1}: {agents[i].name}")
        try_count = 0
        # Assume the user will enter a valid number within 3 tries,
        # otherwise use auto selection to avoid blocking.
        while try_count <= 3:
            try_count += 1
            if try_count >= 3:
                logger.info(f"你已经尝试{try_count}次了。下一个agent将自动选择。")
                break
            try:
                nums = input("输入下一个发言者的序号（不输入任何内容或输入“q”以使用自动选择）：")
                if nums == "" or nums == "q":
                    break
                index = int(nums)
                if index > 0 and index <= _n_agents:
                    return agents[index - 1]
                else:
                    raise ValueError
            except ValueError:
                logger.info(f"输入无效。请输入 1 到 {_n_agents} 之间的数字。")
        return None
```